### ISSUE TITLE
Can't load keys from pagent started manually

### ISSUE THREAD

**Issue by glacjay** [gitextensions/gitextensions#5279]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

- when I start pagent manually (by double clicking pagent.exe file for example), then do some op in GitExtensions that require ssh keys, pagent failed to add the key by reporting: The already running Pageant refused to add the key.
- when I do some op in GitExtensions without start pagent manually first, then pagent will be started by GitExtensions automatically, which is working fine.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

as above

**What is the expected behavior?**

as above

**Environment you encounter the issue:**

- GitExtensions version: 2.51.04
- GIT version: 2.18.0.windows.1
- OS version: windows 10 educational edition
- .NET version: 4.6.1

**Did this work in previous version of GitExtensions (which)?**

not since I noticed

**Comment by RussKie** [comment]
Is there a particular reason for starting the pageant manually?

**Comment by glacjay** [comment]
@RussKie yes, there are other tools can use pagent keys, like WinSCP. Sometimes I start winscp first, and start pagent to use keys in it, and later when I need to use GitExtensions, boom, it doesn't work, then I must exit pagent, and let GitExtensions to start it for me again :-(

**Comment by RussKie** [comment]
Makes sense, thank you for clarification

On Aug 3, 2018 12:12 PM, "GlacJAY" <notifications@github.com> wrote:

@RussKie <https://github.com/RussKie> yes, there are other tools can use
pagent keys, like WinSCP. Sometimes I start winscp first, and start pagent
to use keys in it, and later when I need to use GitExtensions, boom, it
doesn't work, then I must exit pagent, and let GitExtensions to start it
for me again :-(

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/gitextensions/gitextensions/issues/5279#issuecomment-410122987>,
or mute the thread

**Comment by enricoreich** [comment]
Any news on that? Problem still exists with v3.0.2!

**Comment by vbjay** [comment]
Alter ssh settings to use the same pagent that you have running.

**Comment by enricoreich** [comment]
Hello, I checked it and gitextensions is using the same pageant as I use in my logon powershell script which runs in user scope configured though windows logon scripts group policy.
Gitextensions only can access the pagenat key list if it is executed with elevated permissions (administrator). So, every time I use gitextensions and need ssh keys from pageant I have to run gitextensions as administrator OR I have to kill the running pageant process and load the ssh key though gitextensions which spins up pageant and load the selected ssh key without administrative rights and of course without all my other ssh keys.

So, I guess, I have to use gitextensions with elevated permissions (administrator) or is there another option except I already mentioned?

**Comment by RussKie** [comment]
Are GE and pageant running under the same user?

**Comment by enricoreich** [comment]
Yes they do but I figured out that if ps scripts are loaded though user logon GPO, the owner of the spawned process gets changed to administrator even if the owner of the script and application is a normal user. Yeah, windows rights management...its confusing, I "love" it. :D

Now I switched to windows task scheduler and created a task to run the powershell script after login and now it works as it should...without elevated permissions.
Everything is fine with gitextensions but thanks anyway for your help! :-)
